main.py

```diff
@@ -1,86 +1,113 @@
-    
-    for result in model.track(source=source, show=False, stream=True, verbose=False):
+
+    for idx, result in enumerate(model.track(source=source, show=False, stream=True, verbose=False)):
         frame = result.orig_img
-        detections = sv.Detections.from_yolov8(result)
-    
+
+        data = []
+        missing_id = -1
+
+        legitimate_ids = sdu.get_legitimate_ids(id_to_robot_number)
+        current_ids = sdu.get_current_ids(result)
+
+        bad_robots = sdu.get_bad_robot_ids(current_ids, legitimate_ids)
+
+        print('===========Number of iteration: %s===========' % idx)
+
+        if legitimate_ids.issubset(current_ids) and bad_robots:
+            print('False Positives detected %s' % bad_robots)
+            print(f'Current len: {len(current_ids)}')
+
+            data = sdu.remove_false_detections_from_list(data, result)
+            
+            sdu.update_yolo_result(data, result)
+
+            data.clear()
+            bad_robots.clear()
+
+        elif bad_robots:
+            missing_id, bad_id = sdu.find_lost_robot_id(bad_robots, result.boxes.id, id_to_robot_number)
+
+            print("Missing Robot id: %s" % missing_id)
+
+            data = sdu.process_result_list(result, bad_id, missing_id)
+
+            sdu.update_yolo_result(data, result)
+
+            data.clear()
+            bad_robots.clear()
+
+        print(f'Updated ids: {result.boxes.id}')
+
+        detections = sv.Detections.from_ultralytics(result)
+
         if result.boxes.id is not None:
-            detections.tracker_id = sdu.map_tracker_ids(id_to_robot_number, result.boxes.id.cpu().numpy())
-                        
-            bad_robots = set(detections.tracker_id) - set(id_to_robot_number.values())
-                                    
+            detections.tracker_id = sdu.map_tracker_ids(
+                id_to_robot_number, result.boxes.id.numpy())
+
             for detection, tracker_id in zip(detections.xyxy, detections.tracker_id):
-                current_robots.add(tracker_id)
-                                
-                if lost_robots and bad_robots:
-                    print(f'Lost Robot: {lost_robots} Bad Robot: {bad_robots}')
-                    print(f'Bad Robot? {detections.tracker_id[-1]}')
-
-                else:
-                    x1, y1, x2, y2 = detection
-                    
-                    try:
-                        center_x = (x1 + x2) // 2
-                        center_y = (y1 + y2) // 2
-                        
-                        trajectories[tracker_id].append((center_x, center_y))
-                    except:
-                        pass
-                
-            lost_robots = set(id_to_robot_number.values()) - set(current_robots)
-            current_robots.clear()
-                
+                x1, y1, x2, y2 = detection
+
+                try:
+                    center_x = (x1 + x2) // 2
+                    center_y = (y1 + y2) // 2
+
```

Comment:
move to function to the center of a robot

Reply:
I don't know if Liran added such functions, but in my branch I think in utils there is a center function.

Reply:
So should we merge our branches into dev and make this fix there?